File: tasks-api/src/main/java/com/joao/tasks/domain/entity/Collection.java
package com.joao.tasks.domain.entity;

import java.time.LocalDateTime;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.validation.constraints.NotEmpty;

import org.hibernate.validator.constraints.Length;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Collection {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "title")
    @Length(min = 3, max = 50)
    @NotEmpty(message = "Campo título é obrigatório")
    private String title;

    @Column(name = "tasks")
    @OneToMany(fetch = FetchType.EAGER)
    private List<Task> tasks;
    
    @Column(name = "created_date")
    private LocalDateTime createdDate;
}

File: tasks-front/src/app/core/models/collection.ts
import { Task } from "./task";

export interface CollectionType {
    id: number,
    title: string,
    createdDate: string 
}

export class Collection {
    private props: CollectionType = {
        id: 0,
        title: '',
        createdDate: ''
    }
    constructor(obj?: CollectionType){
        if(obj) this.props = obj
    }

    get id(){
        return this.props.id
    }
    
    get title(){
        return this.props.title
    }

    get createdDate(){
        return this.props.createdDate
    }
}